libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/routeprogress/RouteProgressState.java

```diff
@@ -0,0 +1,55 @@
+package com.mapbox.services.android.navigation.v5.routeprogress;
+
+import android.support.annotation.IntDef;
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * Contains the various progress states that can occur while navigating.
+ */
+public final class RouteProgressState {
+
+  private RouteProgressState() {
+    // Class should not be initialized
+  }
+
+  @IntDef({ROUTE_INVALID, ROUTE_INITIALIZED, ROUTE_ARRIVED, LOCATION_TRACKING, LOCATION_STALE})
```

Comment:
should we have OFF ROUTE as well? technically if you dont do a re-route you will stay off route until you get back on